File: 010.py
def isPrime(n):
    r = 2
    while 1:
        if r * r > n:
            return True
        if n % r == 0:
            return False
        r += 1

def oldSln():
    print sum((i for i in xrange(2, 2000 * 1000) if isPrime(i)))


def sieve(m):
    for i in xrange(1, m):

File: 010.hs
-- slow, but work
-- sieve solution, check http://github.com/liuyang1/test/lang/haskell codebase
isPrime x = all (\i -> x `mod` i /= 0) $ takeWhile (\i -> i * i <= x) (2:[3,5..(x `div` 2)])

primes n = filter isPrime [2..n]

main = do
        print $ sum $ primes (1000 * 1000 * 2)
